homeassistant/components/rfxtrx/config_flow.py

```diff
@@ -1,12 +1,281 @@
+                if not device_data.get("event_code"):
+                    errors = {"base": "unknown_event_code"}
+                else:
+                    event_code = device_data["event_code"]
+                    device_id = device_data["device_id"]
+                    self.hass.helpers.dispatcher.async_dispatcher_send(
+                        f"{DOMAIN}_{device_id}"
+                    )
+                    device_registry = await async_get_registry(self.hass)
+                    device_registry.async_remove_device(user_input[CONF_REMOVE_DEVICE])
+                    devices = {event_code: None}
+                    self.update_config_data(
+                        global_options=self._global_options, devices=devices
+                    )
+
+                    return self.async_create_entry(title="", data={})
+            if CONF_DEVICE_ID in user_input:
+                self._selected_device_event_code = user_input[CONF_DEVICE_ID]
+                self._selected_device = {}
+                return await self.async_step_set_device_options()
+
+            if not errors:
+                self.update_config_data(global_options=self._global_options)
+
+                return self.async_create_entry(title="", data={})
+
+        device_registry = await async_get_registry(self.hass)
+        device_entries = async_entries_for_config_entry(
+            device_registry, self._config_entry.entry_id
+        )
+        self._device_entries = device_entries
+
+        devices = {
+            entry.id: entry.name_by_user if entry.name_by_user else entry.name
+            for entry in device_entries
+        }
+
+        options = {
+            vol.Optional(CONF_DEBUG, default=self._config_entry.data[CONF_DEBUG]): bool,
+            vol.Optional(
+                CONF_AUTOMATIC_ADD, default=self._config_entry.data[CONF_AUTOMATIC_ADD],
+            ): bool,
+            vol.Optional(CONF_DEVICE_ID): str,
+            vol.Optional(CONF_DEVICE): vol.In(devices),
+            vol.Optional(CONF_REMOVE_DEVICE): vol.In(devices),
+        }
+
+        return self.async_show_form(
+            step_id="prompt_options", data_schema=vol.Schema(options), errors=errors
+        )
+
+    async def async_step_set_device_options(self, user_input=None):
+        """Manage device options."""
+        errors = {}
+
+        if user_input is not None:
+            device_id = get_device_id(
+                self._rfxobj.device, data_bits=user_input.get(CONF_DATA_BITS)
+            )
+            try:
+                command_on = (
+                    int(user_input.get(CONF_COMMAND_ON), 16)
+                    if user_input.get(CONF_COMMAND_ON)
+                    else None
+                )
+                command_off = (
+                    int(user_input.get(CONF_COMMAND_OFF), 16)
+                    if user_input.get(CONF_COMMAND_OFF)
+                    else None
+                )
```

Comment:
I'm not too fond of those inline if clauses. Something like this seems nicer. Event nicer to use voluptuous to validate.
 
```suggestion
                def none_or_int16(value):
                  if value is None:
                    return None
                  return int(value, 16)
 
                command_on = none_or_int16(user_input.get(CONF_COMMAND_ON))
                command_off = none_or_int16(user_input.get(CONF_COMMAND_OFF))
```

Reply:
I tried voluptuous, no success. Am I doing something wrong?

```
vol.Optional(
   CONF_COMMAND_ON,
   default=hex(device_data.get(CONF_COMMAND_ON, 0)),
   ): vol.All(str, lambda value: int(value, 16))
```

Reply:
The voluptuous-serialize library we use to convert voluptuous for the frontend doesn't support complex validators.